File: backend/src/main/java/dev/mateorossello/merp/modules/accounting/dtos/JournalEntryOutput.java
package dev.mateorossello.merp.modules.accounting.dtos;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

public record JournalEntryOutput(
    Long id,

    LocalDate entryDate,

    LocalDateTime createdAt,

    String description,

    Long createdByUserId,

    List<JournalEntryLineOutput> journalEntryLinesOutput
) {}

File: frontend/src/types/accounting/JournalEntryOutput.ts
import type { JournalEntryLineOutput } from "./JournalEntryLineOutput";

export interface JournalEntryOutput {
  id: number;
  entryDate: string; // ISO 8601 ("AAAA-MM-DD")
  createdAt: string; // ISO 8601 ("AAAA-MM-DDTHH:mm:ss")
  description: string;
  createdByUserId: number;
  journalEntryLinesOutput: JournalEntryLineOutput[];
}
